union.go

```diff
@@ -0,0 +1,58 @@
+package rwvfs
+
+import (
+	"io"
+	pathpkg "path"
+
+	"golang.org/x/tools/godoc/vfs"
+)
+
+// Union returns a new FileSystem which is the union of the provided file systems.
+func Union(fileSystems ...FileSystem) FileSystem {
+	if len(fileSystems) == 0 {
+		return Map(nil)
+	}
+	if len(fileSystems) == 1 {
+		return fileSystems[0]
+	}
+
+	ns := vfs.NameSpace{}
+	for _, fs := range fileSystems {
+		ns.Bind("/", fs, "/", vfs.BindAfter)
+	}
+	return &unionFs{
+		NameSpace:   ns,
+		fileSystems: fileSystems,
+	}
+}
+
+type unionFs struct {
+	vfs.NameSpace
+	fileSystems []FileSystem
+}
+
+func (fs *unionFs) resolve(path string) FileSystem {
+	parent := pathpkg.Dir(path)
+	for _, fs2 := range fs.fileSystems {
+		if _, err := fs2.Stat(parent); err == nil {
```

Comment:
Seems to me that this should check for existence, not just for lack of error. What if the first filesystem returns an error that either (A) indicates that an unrelated error occurred and that the dir may or may not exist, or (B) indicates that the file exists but some other error occurred?

`man fstat` shows a bunch of possible errors:

```
ERRORS
       EACCES Search permission is denied for one of the directories in the path prefix of path.   (See
              also path_resolution(7).)

       EBADF  fd is bad.

       EFAULT Bad address.

       ELOOP  Too many symbolic links encountered while traversing the path.

       ENAMETOOLONG
              path is too long.

       ENOENT A component of path does not exist, or path is an empty string.

       ENOMEM Out of memory (i.e., kernel memory).

       ENOTDIR
              A component of the path prefix of path is not a directory.

       EOVERFLOW
              path  or fd refers to a file whose size, inode number, or number of blocks cannot be rep‐
              resented in, respectively, the types off_t, ino_t, or blkcnt_t.   This  error  can  occur
              when,  for  example,  an  application  compiled on a 32-bit platform without -D_FILE_OFF‐
              SET_BITS=64 calls stat() on a file whose size exceeds (1<<31)-1 bytes.
```

Not all of these indicate that the file doesn't exist. Obviously those don't map 1-to-1 to errors that `Stat` would return, but they're a good starting point.

I am not sure, but I think this might be better behavior:

```go
func (fs *unionFS) resolve(path string) (FileSystem, error) {
  // ...
  if _, err := fs2.Stat(parent); err == nil {
    return fs2, nil
  } else if !os.IsNotExist(err) {
    return nil, err
  }
// ...
}
```

This seems pedantic, but this is such a low-level library that it's worth it IMO.